### PR TITLE
[SAS Token] Add sas token support for v3 jobs.

### PR DESCRIPTION
**Pull request by dannyjdev**
Modified storage account creation to support sas tokens.

### CODE REVIEW

File: src/Ng/Arguments.cs [hunk 1]
```diff
@@ -157,12 +160,14 @@ public static class Arguments
...
         public const string CompressedStorageContainer = "compressedStorageContainer";
         public const string CompressedStorageKeyValue = "compressedStorageKeyValue";
+        public const string CompressedStorageSasValue = "compressedStorageSasValue";
         public const string CompressedStoragePath = "compressedStoragePath";
 
         public const string SemVer2StorageAccountName = "semVer2StorageAccountName";
         public const string SemVer2StorageBaseAddress = "semVer2StorageBaseAddress";
         public const string SemVer2StorageContainer = "semVer2StorageContainer";
         public const string SemVer2StorageKeyValue = "semVer2StorageKeyValue";
+        public const string SemVer2StorageSasValue = "semVer2StorageSasValue";
```
Comment (joelverhagen) [inline src/Ng/Arguments.cs:170]: Unused?

File: src/Ng/Arguments.cs [hunk 2]
```diff
@@ -157,12 +160,14 @@ public static class Arguments
         public const string CompressedStorageBaseAddress = "compressedStorageBaseAddress";
         public const string CompressedStorageContainer = "compressedStorageContainer";
         public const string CompressedStorageKeyValue = "compressedStorageKeyValue";
+        public const string CompressedStorageSasValue = "compressedStorageSasValue";
```
Comment (joelverhagen) [inline src/Ng/Arguments.cs:163]: Unused?

File: src/Ng/Arguments.cs [hunk 3]
```diff
@@ -62,6 +64,7 @@ public static class Arguments
         public const string StorageAccountNameAuditing = "storageAccountNameAuditing";
         public const string StorageContainerAuditing = "storageContainerAuditing";
         public const string StorageKeyValueAuditing = "storageKeyValueAuditing";
+        public const string StorageSasValueAuditing = "storageSasValueAuditing";
```
Comment (joelverhagen) [inline src/Ng/Arguments.cs:67]: I only see this used in help text. Is this parameter actually used?

File: src/Ng/CommandHelpers.cs [hunk 4]
```diff
@@ -151,6 +151,7 @@ public static void AssertAzureStorage(IDictionary<string, string> arguments)
                 { Arguments.StorageBaseAddress, Arguments.StorageBaseAddress + suffix },
                 { Arguments.StorageAccountName, Arguments.StorageAccountName + suffix },
                 { Arguments.StorageKeyValue, Arguments.StorageKeyValue + suffix },
+                { Arguments.StorageSasValue, Arguments.StorageSasValue + suffix },
```
Comment (dannyjdev) [inline src/Ng/CommandHelpers.cs:154]: @joelverhagen the `StorageSasValueAuditing` is used like this. the same goes for `StorageSasValuePreferredPackageSourceStorage`. We have Storage{Sas|Key}Value as prefix to reuse the same method that creates storage account.

Comment (joelverhagen) [reply src/Ng/CommandHelpers.cs:154]: We shouldn't need the `const`s then right? It sounds like they are always auto-generated?

File: src/Ng/Arguments.cs (hunk 1 above)
Comment (dannyjdev) [reply src/Ng/Arguments.cs:170]: This will be used by Ng.Operations job.

File: src/Ng/Arguments.cs (hunk 2 above)
Comment (dannyjdev) [reply src/Ng/Arguments.cs:163]: This will be used by Ng.Operations job.

File: src/Ng/Arguments.cs (hunk 3 above)
Comment (dannyjdev) [reply src/Ng/Arguments.cs:67]: Yes, It's only use there.

File: src/Ng/CommandHelpers.cs (hunk 4 above)
Comment (joelverhagen) [reply src/Ng/CommandHelpers.cs:154]: Okay thanks for clarifying.

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**